### PR TITLE
Added strategies for decoding and encoding maps.

### PR DESCRIPTION
**Pull request by tachyonics**
## Description

Add a transcoding strategy for collapsing/expanding maps from XML in the form-

```xml
<Result>
   <Tag>
     <Key>QueueType</Key>
     <Value>Production</Value>
   </Tag>
   <Tag>
     <Key>Owner</Key>
     <Value>Developer123</Value>
   </Tag>
 </Result>
```

into a Swift structure-

```swift
 struct Result {
     let tags: [String: String]

    enum CodingKeys: String, CodingKey {
        case tags = "Tag"
    }
 }
```

Can also be combined with the collapsing/expanding list strategy if there is an outer tag for the entire list of entries.

## Testing

Added unit tests.

> ./scripts/docker-shortcuts/test.sh
Executed 17 tests, with 0 failures (0 unexpected) in 0.031 (0.031) seconds

### CODE REVIEW

File: Sources/XMLCoding/Decoder/XMLDecoder.swift [hunk 1]
```diff
@@ -216,7 +246,45 @@ internal class _XMLDecoder : Decoder {
...
+        if let currentContainer = self.storage.topContainer as? [String : Any] {
+            
+            // if we are combining collapsing lists and maps
+            if case let .collapseListUsingItemTag(itemTag) = options.listDecodingStrategy,
+                case let .collapseMapUsingTags(keyTag: keyTag, valueTag: valueTag) = options.mapDecodingStrategy,
+                let itemList = currentContainer[itemTag] as? [Any] {
+                    var newContainer: [String: Any] = [:]
+                
+                    // construct a dictionary from each entry and the key and value tags
+                    itemList.forEach { entry in
```
Comment (rafiki270) [inline Sources/XMLCoding/Decoder/XMLDecoder.swift]: this and 278 looks like it could be put in a method to avoid repetitive code possibly?

**Review by rafiki270 — COMMENTED** [review]
looks really good, what do you think about that repeating code?

**Comment by tachyonics** [comment]
Good catch! Fixed to remove duplicated code.

**Comment by tachyonics** [comment]
It looks like the ci is stuck for my second change. It has been in pending for over a day.

**Pull request merged**